### ISSUE TITLE
feat: serde rename support

### ISSUE THREAD

**Issue by lcmgh** [Pantonshire/enumscribe#5]
Hi!

I found that I cannot use `#[serde(rename_all = "snake_case")]` when using enumscribe. Is that something support can be added for?

```rust
#[derive(EnumSerialize, EnumDeserialize, PartialEq, Copy, Debug, Display)]
#[enumscribe(case_insensitive)]
#[serde(rename_all = "snake_case")]
pub enum MyEnum {
    AandB,
    C,
    D,
}
```

BR

**Comment by lcmgh** [comment]
Sorry I think a combination of `snake_case` and case_insensitive does not make really sense because it is mutual exclusive.

**Comment by pantonshire** [comment]
Although I don't intend to try to reimplement every possible serde derive attribute, adding support for `rename_all` seems like a good idea to me.

I do not think renaming a field is mutually-exclusive to matching it case-insensitively; in the example you provided, using both `case_insensitive` and `snake_case` could allow both `"A_AND_B"` and `"a_and_b"` to deserialize to the first variant.

**Comment by pantonshire** [comment]
`rename` and `rename_all` are now implemented in the main branch (a3137d86c2c3af2af584228b036a6c1f6782c77f), I will publish a new release for this soon.

**Comment by pantonshire** [comment]
Version 0.4.0 has been published.
